packages/analytics-controller/src/AnalyticsController.ts

```diff
@@ -0,0 +1,320 @@
+ */
+export type AnalyticsControllerOptions = {
+  state?: Partial<AnalyticsControllerState>;
+  messenger: AnalyticsControllerMessenger;
+  /**
+   * Platform adapter implementation for tracking events
+   */
+  platformAdapter: AnalyticsPlatformAdapter;
+};
+
+/**
+ * The AnalyticsController manages analytics tracking across platforms (Mobile/Extension).
+ * It provides a unified interface for tracking events, identifying users, and managing
+ * analytics preferences while delegating platform-specific implementation to an
+ * {@link AnalyticsPlatformAdapter}.
+ *
+ * This controller follows the MetaMask controller pattern and integrates with the
+ * messenger system to allow other controllers and components to track analytics events.
+ * It delegates platform-specific implementation to an {@link AnalyticsPlatformAdapter}.
+ */
+export class AnalyticsController extends BaseController<
+  'AnalyticsController',
+  AnalyticsControllerState,
+  AnalyticsControllerMessenger
+> {
+  readonly #platformAdapter: AnalyticsPlatformAdapter;
+
+  /**
+   * Constructs an AnalyticsController instance.
+   *
+   * @param options - Controller options
+   * @param options.state - Initial controller state (defaults from getDefaultAnalyticsControllerState)
+   * @param options.messenger - Messenger used to communicate with BaseController
+   * @param options.platformAdapter - Platform adapter implementation for tracking
+   */
+  constructor({
+    state = {},
+    messenger,
+    platformAdapter,
+  }: AnalyticsControllerOptions) {
+    const defaultState = getDefaultAnalyticsControllerState();
+    // Use analyticsId from state if provided, otherwise use the one from defaultState
+    const analyticsId: string =
+      state.analyticsId !== undefined
+        ? state.analyticsId
+        : defaultState.analyticsId;
+    const mergedState: AnalyticsControllerState = {
+      ...defaultState,
+      ...state,
+      analyticsId,
+    };
+
+    super({
+      name: controllerName,
+      metadata: analyticsControllerMetadata,
+      state: mergedState,
+      messenger,
+    });
+
+    this.#platformAdapter = platformAdapter;
+
+    this.messenger.registerMethodActionHandlers(
+      this,
+      MESSENGER_EXPOSED_METHODS,
+    );
+
+    projectLogger('AnalyticsController initialized and ready', {
+      enabled: this.state.enabled,
+      optedIn: this.state.optedIn,
+      analyticsId: this.state.analyticsId,
+    });
+  }
+
+  /**
+   * Track an analytics event.
+   *
+   * Events are only tracked if analytics is enabled.
+   *
+   * @param eventName - The name of the event
+   * @param properties - Event properties
+   */
+  trackEvent(
+    eventName: string,
+    properties: AnalyticsEventProperties = {},
+  ): void {
+    // Don't track if analytics is disabled
+    if (!this.state.enabled) {
+      return;
+    }
+
+    // Delegate to platform adapter
+    this.#platformAdapter.trackEvent(eventName, properties);
```

Comment:
Can't we get closer to the segment adapter itself rather than having another layer of abstraction on top ?

Reply:
My thinking is that we shouldn't need three layers of abstraction around Segment in the clients.

Today we have custom classes created to interface with Segment:
- https://github.com/MetaMask/metamask-extension/blob/d4ef57f56992fac2579765a01c34b7f4dda5e2b0/app/scripts/lib/segment/analytics.js
- https://github.com/MetaMask/metamask-mobile/blob/1c7b545ec2a6c713129aae9ce1cebf9fbbeddef5/app/core/Analytics/MetaMetrics.ts#L591

I think it would be easier if we could remove the need of another layer of abstraction on top of these utils by just using the Segment classes directly, unifying the signature of them rather than adding another class to wrap them.